### DISCUSSION TITLE
(Version 2 ) API Inside form to preform some actions

### DISCUSSION THREAD

**Discussion by Hegabovic** [filamentphp/filament#8251] [Help] [answered]
is there a way to call an endpoint in edit form field to be able to preform some logic 

for example i want to call a get request which will populate a drop-down list then i will be able to choose from it

**Comment by leandrocfe** [comment, marked as answer]
This?

```php
//use Illuminate\Support\Facades\Http;
Select::make('select')
    ->options(Http::get('https://dummyjson.com/products/categories')->json())
```
You can also use this

```php
Select::make('select')
    ->options(function (): array {

        //your logic...
        $options = [
            '1' => 'Option 1',
            '2' => 'Option 2',
        ];

        return $options;
    })
```

**Reply by Hegabovic** [reply]
can i enter search key for the same field for example 

i'm calling auto complete weather api 
i want to enter something like alexandria and then a call is made on click to retrieve a drop down list which i will choose from to get the key i want and save it to database 

i want to use this because there is a lot of customization on this endpoint 
```
Select::make('select')
    ->options(function (): array {

        //your logic...
        $options = [
            '1' => 'Option 1',
            '2' => 'Option 2',
        ];

        return $options;
    })
```
for example 
![Screenshot from 2023-09-03 15-09-55](https://github.com/filamentphp/filament/assets/56083879/a36aaaad-79bd-495b-b4b5-012a10863542)

i want to be able to enter Alexandria and the drop-down list is generated based on my entry

**Reply by leandrocfe** [reply]
I think you can use `getSearchResultsUsing`
```php
->getSearchResultsUsing(function (string $search): array {
    //your logic...
    $result = [];
    return $result;
})
```
https://filamentphp.com/docs/3.x/forms/fields/select#returning-custom-search-results
